### ISSUE TITLE
Will it support Android 9.0?

### ISSUE THREAD

**Issue by Nightsoulyi** [noname8964/vold-posix#5]
I'm using android 9.0,it has some problems about mount file system,I need this tool to deal with it.
Please help me.Thank you.

**Comment by noname8964** [comment]
Will be in the future. Waiting for pie update for my phone.
and you can compile it yourself atm. just using ``repo`` clone the pie source code from AOSP and replace $SOURCE/system/vold to my fork and clone exfat/ntfs-3g/fuse to $SOURCE/external then compile them.

**Comment by wuchjun** [comment]
> Will be in the future. Waiting for pie update for my phone.
> and you can compile it yourself atm. just using `repo` clone the pie source code from AOSP and replace $SOURCE/system/vold to my fork and clone exfat/ntfs-3g/fuse to $SOURCE/external then compile them.
Could I just replase with the ready-made files in other roms of my device?

**Comment by null4n** [comment]
nope, only fuse driver can. My vold has modified for exfat, so if you do
that, exfat will not work.

**Comment by AuroraWright** [comment]
I made a 9.0 version (I think it just supports Treble devices though, it depends if they use .cil files for the sepolicy). For the exFAT support I went with LineageOS 16's decision and kept the built-in support (so you still need to compile sdFAT into your device's kernel). See here https://forum.xda-developers.com/pixel-2-xl/development/mod-ext4-exfat-ntfs-support-stock-pie-t3881473 .
